suzieq/engines/pandas/network.py

```diff
@@ -204,20 +209,33 @@ def _find_addr_arp(self, addr: str, **kwargs) -> pd.DataFrame:
         # that precludes us addresses that were not remote in the past.
         macdf = self._get_table_sqobj('macs', start_time='') \
                     .get(namespace=namespaces, hostname=hostnames,
-                         macaddr=macaddrs, columns=['default'])
+                         macaddr=macaddrs,
+                         columns=['namespace', 'hostname', 'vlan', 'macaddr',
+                                  'oif', 'remoteVtepIp', 'flags', 'active',
+                                  'timestamp'])
 
         for arp_row in arpdf.itertuples():
-            row_end_time = arp_row.timestamp + timedelta(seconds=60)
+            row_end_time = arp_row.next_ts or self.iobj.end_time
             tmpres = {}
             if getattr(arp_row, 'error', None):
                 continue
             if arp_row.oif.endswith('-v0'):
-                # Handle VRR interfaces in Cumulus
-                oif = arp_row.oif[:-3]
-            else:
-                oif = arp_row.oif
+                # Ignore VRR interfaces in Cumulus, we got them via the
+                # normal interface entry (without -v0)
+                continue
+            oif = arp_row.oif
 
             active = getattr(arp_row, 'active', True)
+            if (arp_row.remote and
+                    arp_row.hostname not in active_local_hosts):
+                continue
```

Comment:
Can you add a comment here? Now we know what this 'if' does, but in the future we'll forget.

Reply:
OK